File: livingrimoire start here/LivinGrimoire java/src/Auxiliary_Modules/TimedMessages.java
package Auxiliary_Modules;

import java.util.HashMap;
import java.util.Map;
import java.util.Random;

public class TimedMessages {
    public Map<String, String> messages = new HashMap<>();
    private String lastMSG = "nothing";
    private Boolean msg = false;
    public void addMSG(String ear){
        String tempMSG = RegexUtil.extractRegex("(?<=remind me to).*?(?=at)",ear);
        if(tempMSG.isEmpty()){return;}
        String timeStamp = RegexUtil.extractRegex(enumRegexGrimoire.simpleTimeStamp,ear);
        if(timeStamp.isEmpty()){return;}
        messages.put(timeStamp, tempMSG);
    }
    public void addMSGV2(String timeStamp, String msg){
        messages.put(timeStamp, msg);
    }
    public void sprinkleMSG(String msg, int amount){
        for (int i = 0; i < amount; i++) {
            messages.put(generateRandomTimestamp(), msg);
        }
    }
    public static String generateRandomTimestamp() {
        // Generate a random number of minutes (0 to 59)
        Random random = new Random();
        int minutes = random.nextInt(60);
        String m;
        if(minutes>9){
            m = String.format("%d", minutes);
        }else{
            m = String.format("0%d", minutes);
        }
        int hours = random.nextInt(12);
        if(hours>9){
            return String.format("%d:%s", hours, m);
        }
        return String.format("0%d:%s", hours, m);
    }
    public void clear(){messages = new HashMap<>();}
    public void tick(){
        String now = TimeUtils.getCurrentTimeStamp();
        if(this.messages.containsKey(now)){
            if(!lastMSG.equals(messages.get(now))){
                lastMSG = messages.get(now);
                msg = true;
            }
        }
    }

    public String getLastMSG() {
        msg = false;
        return lastMSG;
    }

    public Boolean getMsg() {
        return msg;
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/auxiliary_modules/TimedMessages.kt
package auxiliary_modules

import java.util.*

class TimedMessages {
    var messages: MutableMap<String, String> = HashMap()
    private var lastMSG: String = "nothing"
    var msg = false
        private set

    fun addMSG(ear: String) {
        val tempMSG = RegexUtil.extractRegex("(?<=remind me to).*?(?=at)", ear)
        if (tempMSG.isEmpty()) {
            return
        }
        val timeStamp: String = RegexUtil.extractRegex(enumRegexGrimoire.SimpleTimeStamp, ear)
        if (timeStamp.isEmpty()) {
            return
        }
        messages[timeStamp] = tempMSG
    }

    fun addMSGV2(timeStamp: String, msg: String) {
        messages[timeStamp] = msg
    }

    fun sprinkleMSG(msg: String, amount: Int) {
        for (i in 0 until amount) {
            messages[generateRandomTimestamp()] = msg
        }
    }

    fun clear() {
        messages = HashMap()
    }

    fun tick() {
        val now: String = TimeUtils.currentTimeStamp
        if (messages.containsKey(now)) {
            if (lastMSG != messages[now]) {
                lastMSG = messages[now]!!
                msg = true
            }
        }
    }

    fun getLastMSG(): String {
        msg = false
        return lastMSG
    }

    companion object {
        fun generateRandomTimestamp(): String {
            // Generate a random number of minutes (0 to 59)
            val random = Random()
            val minutes = random.nextInt(60)
            var m = ""
            m = if (minutes > 9) {
                String.format("%d", minutes)
            } else {
                String.format("0%d", minutes)
            }
            val hours = random.nextInt(12)
            return if (hours > 9) {
                String.format("%d:%s", hours, minutes)
            } else String.format("0%d:%s", hours, minutes)
        }
    }
}